pallets/nft/src/lib.rs

```diff
@@ -66,6 +69,18 @@ pub struct ClassData<BN, ID> {
 	pub end_block: Option<BN>,
 	/// merged from two class; if true, burn the two items 
 	pub class_type: ClassType<ID>,
+  /// claimed vec for claim type NFT class, to guarantee each user claims once 
+  claimed_vec: Vec<u32>, // maximal index of claiming user is 2^32 which is more than enough
+}
+
+impl<BN, ID> ClassData<BN, ID> {
+  fn is_claimed(&self, index: u32) -> bool {
+    self.claimed_vec.contains(&index)
+  }
+
+  fn record_claim(&mut self, index: u32) {
```

Comment:
Don't we check wether this is already claimed? It seems to easy to just `push` the index.